core/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -940,7 +937,7 @@ public String cause() {
          * @return snapshot name
          */
         public String name() {
-            return name;
+            return snapshot.getName();
```

Comment:
we should probably expose snapshot directly as I see that usage points create a new Snapshot(..) again based on the data here...